File: JavaFundamentals/JavaAdvanced/LinearDataStructures_Ex/MatrixOfPalindromes.java
import java.util.Scanner;

public class MatrixOfPalindromes {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int r = scanner.nextInt();
        int c = scanner.nextInt();
        String[][] matrix = new String[r][c];

        for (int row = 0; row < r; row++) {
            for (int col = 0; col < c; col++) {
                char side = (char)('a' + row);
                char center = (char)('a' + row + col);
                String value = "" + side + center + side;
                matrix[row][col] = value;
            }
        }

        for (int row = 0; row < r; row++) {
            for (int col = 0; col < c; col++) {
                System.out.print(matrix[row][col] + " ");
            }
            System.out.println();
        }
    }
}

File: CSharpFundamentals/CSharpAdvanced/Matrix_Ex/MatrixOfPalindromes.cs
using System;

namespace MatrixOfPalindromes
{
    class MatrixOfPalindromes
    {
        private static string[,] matrix;
        static void Main(string[] args)
        {
            ReadMatrix();
            for (int row = 0; row < matrix.GetLength(0); row++)
            {
                for (int col = 0; col < matrix.GetLength(1); col++)
                {
                    Console.Write(matrix[row, col] + " ");
                }
                Console.WriteLine();
            }
        }

        private static void ReadMatrix()
        {
            string[] sizes = Console.ReadLine().Split();
            int rows = int.Parse(sizes[0]);
            int cols = int.Parse(sizes[1]);
            char[] alphabet = "abcdefghijklmnopqrstuvwxyz".ToCharArray();
            matrix = new string[rows, cols];
            for (int row = 0; row < rows; row++)
            {
                for (int col = 0; col < cols; col++)
                {
                    char a = alphabet[row];
                    char b = alphabet[row + col];
                    char[] arr = new char[3] { a, b, a };
                    matrix[row, col] = new string(arr);
                }
            }
        }
    }
}
